qiskit/transpiler/__init__.py

```diff
@@ -40,714 +41,1132 @@
+   qc_basis = transpile(qc, backend)
+   qc_basis.draw(output='mpl')
 
-      qc = QuantumCircuit(2, 1)
+A few things to highlight. First, the circuit has gotten longer with respect to the
+initial one.  This can be verified by checking the depth of the circuits:
 
-      qc.h(0)
-      qc.x(1)
-      qc.cp(np.pi/4, 0, 1)
-      qc.h(0)
-      qc.measure([0], [0])
-      qc.draw(output='mpl')
+.. code-block::
 
-      backend = FakeVigoV2()
+   print('Original depth:', qc.depth(), 'Decomposed Depth:', qc_basis.depth())
 
-      qc_basis = transpile(qc, backend)
-      qc_basis.draw(output='mpl')
+.. parsed-literal::
 
+    Original depth: 4 Decomposed Depth: 10
 
-   A few things to highlight.  First, the circuit has gotten longer with respect to the
-   initial one.  This can be verified by checking the depth of the circuits:
+Second, although we had a single controlled gate, the fact that it was not in the basis
+set means that, when expanded, it requires more than a single :class:`~.CXGate` to implement.
+All said, unrolling to the basis set of gates leads to an increase in the depth of a
+quantum circuit and the number of gates.
 
-   .. code-block::
+It is important to highlight two special cases:
 
-      print('Original depth:', qc.depth(), 'Decomposed Depth:', qc_basis.depth())
+1. If A swap gate is not a native gate and must be decomposed this requires three CNOT gates:
+   .. code-block::
 
-   .. parsed-literal::
+      from qiskit.providers.fake_provider import FakeVigoV2
+      backend = FakeVigoV2()
 
-       Original depth: 4 Decomposed Depth: 10
+      print(backend.operation_names)
 
-   Second, although we had a single controlled gate, the fact that it was not in the basis
-   set means that, when expanded, it requires more than a single `cx` gate to implement.
-   All said, unrolling to the basis set of gates leads to an increase in the depth of a
-   quantum circuit and the number of gates.
+   .. parsed-literal::
 
-   It is important to highlight two special cases:
+      ['id', 'rz', 'sx', 'x', 'cx', 'measure', 'delay']
 
-   1. A SWAP gate is not a native gate on the IBM Q devices, and must be decomposed into
-      three CNOT gates:
+   .. plot:
+      :include-source:
 
-      .. plot::
-         :include-source:
+      from qiskit.circuit import QuantumCircuit
 
-         from qiskit import QuantumCircuit
+      swap_circ = QuantumCircuit(2)
+      swap_circ.swap(0, 1)
+      swap_circ.decompose().draw(output='mpl')
 
-         swap_circ = QuantumCircuit(2)
-         swap_circ.swap(0, 1)
-         swap_circ.decompose().draw(output='mpl')
+   As a product of three CNOT gates, swap gates are expensive operations to perform on a
+   noisy quantum devices.  However, such operations are usually necessary for embedding a
+   circuit into the limited entangling gate connectivities of actual devices. Thus,
+   minimizing the number of swap gates in a circuit is a primary goal in the
+   transpilation process.
```

Comment:
```suggestion
   Swap gates are expensive operations to perform on a noisy quantum devices because
   each one is decomposed to three CNOT gates.  However, swaps are usually necessary
   for embedding a circuit to match the limited connectivity of actual devices.
   Thus, a primary goal of the transpilation process is to minimize the number of swap gates
   required to make the circuit compatible with the target device.
```